src/SetHo.hs

```diff
@@ -39,8 +39,8 @@ defaultSetHoConfig =
   }
 
 -- | fire up the the GUI
-runSetter :: Maybe SetHoConfig -> String -> DTree -> IO (Maybe (Int, DTree)) -> (Int -> IO ()) -> (Int -> DTree -> IO ()) -> IO ()
-runSetter mconfig rootName initialValue userPollForNewMessage sendRequest userCommit = do
+runSetter :: Maybe SetHoConfig -> String -> DTree -> IO (Maybe (Int, DTree)) -> (Int -> IO ()) -> (Int -> DTree -> IO ()) -> (Int -> DTree -> IO ()) -> IO ()
```

Comment:
This breaks the binary in the examples directory, can you please update it?

Reply:
I'm unable to figure out with Cabal how to actually build this example.  I nixified all the dependencies and can build the library, but:
```
peddie@jaidee:Plot-ho-matic(master☠)$ nix-shell --pure shell.nix --run "ghcid --command 'cabal v2-build set-example'"
Loading cabal v2-build set-example ...
Resolving dependencies...
Build profile: -w ghc-8.10.4 -O1
In order, the following will be built (use -v for more details):
 - Plot-ho-matic-0.12.2.3 (exe:set-example) (configuration changed)
Configuring executable 'set-example' for Plot-ho-matic-0.12.2.3..
Warning: The package has an extraneous version range for a dependency on an
internal library: Plot-ho-matic >=0 && ==0.12.2.3, Plot-ho-matic >=0 &&
==0.12.2.3. This version range includes the current package but isn't needed
as the current package's library will always be used.
Preprocessing executable 'set-example' for Plot-ho-matic-0.12.2.3..
Building executable 'set-example' for Plot-ho-matic-0.12.2.3..
Command "cabal v2-build set-example" exited unexpectedly with error message: as the current package's library will always be used.
```
The incomplete error message, combined with the problematic version bounds on `Plot-ho-matic` which appear nowhere in the actual `.cabal` file, make me think this might be a cabal bug (shocker).  Removing `Plot-ho-matic` from the `build-depends:` field of the executable does not solve the problem.

Reply:
Anyway, I think I fixed the example, but I can't check that it compiles thanks to cabal.

Reply:
I checked this with stack as recommended after rebasing on `master`, and it works.